accounts/views.py

```diff
@@ -8,36 +8,34 @@
 from django.shortcuts import redirect
 from django.utils.decorators import method_decorator
 from django.views.decorators.csrf import csrf_exempt
+from django.views.generic.base import View
 from oauth2_provider.models import get_access_token_model
 from oauth2_provider.views import IntrospectTokenView
-from rest_framework import generics
-from rest_framework_api_key.permissions import HasAPIKey
 from sentry_sdk import capture_message
 
 from accounts.auth import LabsView, PennView
 from accounts.serializers import UserSerializer
 
 
-class LoginView(generics.GenericAPIView):
+class LoginView(View):
     """
     Log in a user.
+    WARNING: You must ensure this page is protected by Shibboleth and Clean Headers
+    See https://github.com/nginx-shib/nginx-http-shibboleth
     """
-    def get(self, request):
-        # Validate API Key
-        if not HasAPIKey.has_permission(self, request, self):
-            return redirect('https://auth.pennlabs.org/login/?next=' + request.GET.get('next', ''))
 
-        # API is valid, login user
+    def get(self, request):
         pennkey = request.META.get('HTTP_EPPN', '').lower().split('@')[0]
         first_name = request.META.get('HTTP_GIVENNAME', '').lower().capitalize()
         last_name = request.META.get('HTTP_SN', '').lower().capitalize()
         email = request.META.get('HTTP_MAIL', '').lower()
-        shibboleth_attributes = {'first_name': first_name, 'last_name': last_name, 'email': email}
+        affiliation = request.META.get('HTTP_UNSCOPED_AFFILIATION', '').split(';')
+        shibboleth_attributes = {'first_name': first_name, 'last_name': last_name, 'email': email,
```

Comment:
From my comment above, looks like shibboleth_attributes being None would mean that something went wrong.

Reply:
That's true. I added the check since I had previously written a test where `shibboleth_attributes` was empty, but you're right, it would be impossible for that to happen.

Reply:
Fixed in b75f0afb3002015b0c19c7ba7c958da7718e0155